addons/point_of_sale/static/src/app/screens/ticket_screen/ticket_screen.js

```diff
@@ -112,6 +114,24 @@ export class TicketScreen extends Component {
             BACKSPACE,
         ]);
     }
+    onClickScanOrder(qrcode) {
+        if (qrcode) {
+            const ref = new URL(qrcode).searchParams.get("order_uuid");
+            const order = this.pos.models["pos.order"].getBy("uuid", ref);
```

Comment:
not enough, you should look at the server records as well. `onClickScanOrder` should be async to allow waiting for server call.